spring-data-mongodb/src/main/java/org/springframework/data/mongodb/core/ChangeStreamEvent.java

```diff
@@ -157,17 +165,38 @@ public T getBody() {
 			return targetType.cast(fullDocument);
 		}
 
-		return getConverted(fullDocument);
+		return getConvertedFullDocument(fullDocument);
+	}
+
+	@Nullable
+	public T getBodyBeforeChange() {
+
+		if (raw == null) {
+			return null;
+		}
+
+		Document fullDocumentBeforeChange = raw.getFullDocumentBeforeChange();
```

Comment:
Nit: Could be combined with `if(raw == null || raw.getFullDocumentBeforeChange() == null)`

Reply:
Fixed